bundles/http/Citrix/SAP-Concur-Demo-HTTP/script.js

```diff
@@ -0,0 +1,605 @@
+        CashAdvanceApplied: false, 
+        Amount: 600.0000, 
+        CurrencyCode: 'USD', 
+        ReportId: '12372882'
+        },
+        {
+        Id: '19710704732971819', 
+        Vendor: 'Starwood Hotels', 
+        SpendingCategory: 'Lodging',
+        TransactionDate: fiveDaysAgo, 
+        CashAdvanceApplied: false, 
+        Amount: 200.0000, 
+        CurrencyCode: 'USD', 
+        ReportId: '12381535'
+        }
+  ]
+
+const categories = [
+  {
+    Type: 'Ground Transportation',
+    ImageURL: 'https://iws-stage-global-cdn-endpoint.azureedge.net/microapps/assets/exported/meeting.a44eb1e81b027ba58ea71078394ca1a0.svg'
+  },
+  {
+    Type: 'Lodging',
+    ImageURL: 'https://iws-stage-global-cdn-endpoint.azureedge.net/microapps/assets/exported/account.7e6a4a82a354a98226bc0c0b4ec9039b.svg'
+  },
+  {
+    Type: 'Airline Fee',
+    ImageURL: 'https://iws-stage-global-cdn-endpoint.azureedge.net/microapps/assets/exported/itinerary.82ae0728d114f607d5b618e3b4eba614.svg'
+  },
+  {
+    Type: 'Meal',
+    ImageURL: 'https://iws-stage-global-cdn-endpoint.azureedge.net/microapps/assets/exported/change_request.b0ee4a80fb3ec8b889c043706b26bb1d.svg'
+  },
+  {
+    Type: 'Entertainment',
+    ImageURL: 'https://iws-stage-global-cdn-endpoint.azureedge.net/microapps/assets/exported/default-broken.b3426980ecaefcfc0778d475c1fe4dd6.svg'
+  },
+  {
+    Type: 'Other',
+    ImageURL: 'https://iws-stage-global-cdn-endpoint.azureedge.net/microapps/assets/exported/feedback.c533ed2d280d39d9288b2fd741a18651.svg'
+  },
+  {
+    Type: 'Office Supplies',
+    ImageURL: 'https://iws-stage-global-cdn-endpoint.azureedge.net/microapps/assets/exported/case.2bb73f1b3eff52a8134fd591447bdaa1.svg'
+  },
+  {
+    Type: 'Advertising/Marketing',
+    ImageURL: 'https://iws-stage-global-cdn-endpoint.azureedge.net/microapps/assets/exported/lead.e61e72321674c9469b115c0552c5dc65.svg'
+  },
+  {
+    Type: 'Office Software',
+    ImageURL: 'https://iws-stage-global-cdn-endpoint.azureedge.net/microapps/assets/exported/report_update.2d453d1646665c2ede6de83f848a6566.svg'
+  }
+]
+
+async function syncReports ({ dataStore, integrationParameters }) {
+  console.log('Synching data...')
+
+  let saveReports = calculateReportsWithTotals(expenses, reports)
+
+  dataStore.save('reports', saveReports)
+  dataStore.save('expenses', expenses)
+  dataStore.save('categories', categories)
+  console.log("Synchronization complete")
+}
+
+function deepClone(list) {
+  return JSON.parse(JSON.stringify(list[generateRandomNumValue(0, list.length)]))
+}
+
+function generateRandomNumValue(floor, ceiling) {
+  return Math.floor(Math.random()* (ceiling - floor)) + floor
+}
+
+function calculateReportsWithTotals(expenses, reports) {
+  let amounts = expenses.reduce((acc, {ReportId, Amount}) => {    
+    acc[ReportId] = (acc[ReportId] || 0) + Amount
+    return acc
+    }, {} )
+	//is used exclusively for incremental sync, since we're passing a single object here--so map() won't work
```

Comment:
I'm not setting total amount to an array. It's just that "reports" could be either an object (one report) or an array of reports. I will change the function to have a third parameter, boolean, that will tell the function whether report is a list of reports or just one report and make the condition more readable. Will that be better?

Reply:
> third parameter, boolean

it's a bad practice to extend function responsibility by flags. Solution is simple, 1) remove

```
if (Object.keys(amounts).length===1) {
    reports.TotalAmount = amounts[reports.Id]
    return
}
```
and on the line 591 call it `const reports = calculateReportsWithTotals(newExpenses, [report])` and then save the result `reports` instead of input argument `report`